be/src/storage/sstable/format.cpp

```diff
@@ -66,6 +67,42 @@ Status Footer::DecodeFrom(Slice* input) {
     return result;
 }
 
+/**
+ * Decompress the input data using the specified compression algorithm.
+ * @param compression_type The compression algorithm to use.
+ * @param input The input data to decompress.
+ * @param length The length of the input data.
+ * @param result The decompressed data.
+ * @return Status::OK() if the decompression was successful, an error otherwise.
+ */
+static Status DecompressBlock(CompressionTypePB compression_type, const char* input, size_t length,
+                              BlockContents* result) {
+    const BlockCompressionCodec* codec = nullptr;
+    // Choose the compression algorithm that we need.
+    RETURN_IF_ERROR(get_block_compression_codec(compression_type, &codec));
+
+    // Get decompressed length from the first 64 bits of the input.
+    uint64_t output_length = DecodeFixed64(input);
+
+    // Allocate memory for the decompressed data.
+    char* output = new char[output_length];
+
+    // Prepare a Slice to hold the decompressed data.
+    Slice output_slice(output, output_length);
+
+    // Prepare a Slice for the compressed data, excluding the first 64 bits used for the length.
+    Slice input_slice(input + sizeof(uint64_t), length - sizeof(uint64_t));
+
+    // Decompress the input data into the output slice.
+    RETURN_IF_ERROR(codec->decompress(input_slice, &output_slice));
```

Comment:
memory leak if error here, the `output` memory is not freed

Reply:
fix